results/pkg/watcher/reconciler/controller.go

```diff
@@ -59,60 +59,94 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
-		return nil
+		return err
 	}
 
-	// Get the Task Run resource with this namespace/name
-	tr, err := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		// The resource no longer exists, in which case we stop processing.
-		r.logger.Infof("task run %q in work queue no longer exists", key)
-		return nil
-	} else if err != nil {
-		r.logger.Errorf("Error retrieving Result %q: %s", name, err)
+	// Get the resource object with this namespace/name
+	taskRun, terr := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
+	pipelineRun, perr := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
+
+	if errors.IsNotFound(terr) && errors.IsNotFound(perr) {
+		r.logger.Infof("Task/Pipeline run %q in work queue no longer exists", key)
 		return err
 	}
+	if errors.IsNotFound(perr) && terr != nil {
+		r.logger.Errorf("Error retrieving TaskRun %q: %s", name, terr)
+		return terr
+	}
+	if errors.IsNotFound(terr) && perr != nil {
+		r.logger.Errorf("Error retrieving PipelineRun %q: %s", name, perr)
+		return perr
+	}
 
-	r.logger.Infof("Receiving new Result %s/%s", namespace, tr.Name)
+	var result *pb.Result = nil
 
-	// Send the new status of the Result to the API server.
-	p, err := convert.ToProto(tr)
-	if err != nil {
-		r.logger.Errorf("Error converting to proto: %v", err)
-		return err
+	val := ""
+	ok := false
+	resultType := "OPAQUE"
+
+	if taskRun != nil {
+		resultType = "TASKRUN"
+		r.logger.Infof("Receiving new Result(%s) %s/%s", resultType, namespace, name)
+		p, err := convert.ToTaskRunProto(taskRun)
+		if err != nil {
+			r.logger.Errorf("Error converting to proto: %v", err)
+			return err
+		}
+		result = &pb.Result{
+			Executions: []*pb.Execution{{
+				Execution: &pb.Execution_TaskRun{p},
+			}},
+		}
+		val, ok = p.GetMetadata().GetAnnotations()[idName]
 	}
-	res := &pb.Result{
-		Executions: []*pb.Execution{{
-			Execution: &pb.Execution_TaskRun{p},
-		}},
+
+	if pipelineRun != nil {
+		resultType = "PIPELINERUN"
+		r.logger.Infof("Receiving new Result(%s) %s/%s", resultType, namespace, name)
+		p, err := convert.ToPipelineRunProto(pipelineRun)
+		if err != nil {
+			r.logger.Errorf("Error converting to proto: %v", err)
+			return err
+		}
+		result = &pb.Result{
+			Executions: []*pb.Execution{{
+				Execution: &pb.Execution_PipelineRun{p},
+			}},
+		}
+		val, ok = p.GetMetadata().GetAnnotations()[idName]
 	}
 
-	// Create a Result if it does not exist in results server, update existing one otherwise.
-	if val, ok := p.GetMetadata().GetAnnotations()[idName]; ok {
-		res.Name = val
+	if ok {
+		result.Name = val
 		if _, err := r.client.UpdateResult(ctx, &pb.UpdateResultRequest{
 			Name:   val,
-			Result: res,
+			Result: result,
 		}); err != nil {
-			r.logger.Errorf("Error updating TaskRun %s: %v", name, err)
+			r.logger.Errorf("Error updating Tekton Result(%s): %s\n%v", resultType, val, err)
```

Comment:
We can use zap to automatically tag all logs with theses values - https://pkg.go.dev/go.uber.org/zap#Logger.With e.g.

```go
log := r.logger.With(zap.String("type", "PipelineRun"), zap.String("result", val))
log.Errorf(...)
```